application/src/main/java/org/togetherjava/tjbot/features/utils/LinkDetections.java

```diff
@@ -0,0 +1,69 @@
+package org.togetherjava.tjbot.features.utils;
+
+import com.linkedin.urls.Url;
+import com.linkedin.urls.detection.UrlDetector;
+import com.linkedin.urls.detection.UrlDetectorOptions;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Utility class to detect links.
+ */
+public class LinkDetections {
+
+    private LinkDetections() {
+        throw new UnsupportedOperationException("Utility class");
+    }
+
+    /**
+     * Extracts all links from the given content.
+     *
+     * @param content the content to search through
+     * @param filterSuppressed filters links suppressed with {@literal <url>}
+     * @param filterNonHttpSchemes filters links that are not using http scheme
+     * @return a list of all found links, can be empty
+     */
+    public static List<String> extractLinks(String content, boolean filterSuppressed,
+            boolean filterNonHttpSchemes) {
```

Comment:
im putting this as general note, cause its not super important for now, but this is the so called "boolean trap".

to a reader without param-hints, `extractLinks(text, true, false)` is confusing, as they have no idea what the bools mean (without looking up the doc).

a code-way to fix the issue is by introducing quick enums instead. like

```java
enum LinkFilter {
  SUPPRESSED,
  NON_HTTP_SCHEME
}
```
and then instead accept a `Set<LinkFilter> filters` as parameter. the call-site then looks like
```java
extractLinks(text, Set.of(LinkFilter.SUPPRESSED, ...))
```

Reply:
I knew you would say that.
Thanks for the suggestion, somehow couldn't think of the enum solution myself lol
Would it maybe make more sense using varargs for the parameter?

Reply:
> im putting this as general note, cause its not super important for now, but this is the so called "boolean trap".
> 
> to a reader without param-hints, `extractLinks(text, true, false)` is confusing, as they have no idea what the bools mean (without looking up the doc).
> 
> a code-way to fix the issue is by introducing quick enums instead. like
> 
> ```java
> enum LinkFilter {
>   SUPPRESSED,
>   NON_HTTP_SCHEME
> }
> ```
> and then instead accept a `Set<LinkFilter> filters` as parameter. the call-site then looks like
> ```java
> extractLinks(text, Set.of(LinkFilter.SUPPRESSED, ...))
> ```

Yea I had some difficulty catching up with flags but then since there was documentation as well, it was okayish for me.

Reply:
im not a fan of varargs cause of all the edge cases u would have to handle. In this case:
* null
* duplicates

and `Set.of` is kinda convenient already anyways, fortunately.

but sure, its not a big deal for now